api/definitions/v1.buildsource/description.adoc

```diff
@@ -0,0 +1 @@
+A BuildSource is an object describing the SCM used for the <<v1.Build>>. A BuildSource contains a type string which defines the type of SCM used for the build. Currently, the only SCM supported by OpenShift is Git. As such, the BuildSource contains a <<v1.GitBuildSource>> to hold optional additional information about the Git buid source. The BuildSource also contains a string that defines the sub-directory in the SCM repository where the source code for the application exists. This allows to have buildable sources in directory other than root of a repository. The BuildSource also contains the source `Secret` that should be used for setting up the authentication for cloning private repository. The secret contains valid credentials for remote repository, where the data's key represent the authentication method to be used and value is the base64 encoded credentials. Supported auth methods are: ssh-privatekey.
```

Comment:
> ...to hold optional additional information about the Git buid source

to hold optional additional information about the Git *build* source

> The BuildSource also contains a string that defines the sub-directory in the SCM repository where the source code for the application exists. This allows to have buildable sources in directory other than root of a repository. 

The contextDir in the BuildSource can specify a sub-directory in the SCM repository to use for the build.

> for setting up the authentication for cloning private repository

for setting up the authentication for cloning *a* private repository

> The secret contains valid credentials for remote repository, where the data's key represent the authentication method to be used and value is the base64 encoded credentials. Supported auth methods are: ssh-privatekey.

The secret contains valid credentials for *the* remote source repository. The secret must contain a data key with the name of the authentication method to use and the credentials in base64 encoded form in the value. Currently, OpenShift only supports `ssh-privatekey` as an authentication method.